src/nvim/os/shell.c

```diff
@@ -37,6 +37,50 @@ typedef struct {
 #endif
 
 
+/// Process command string with 'shellxescape' (p_sxe) and 'shellxquote'
+/// (p_sxq)
+///
+/// @param cmd Command string
+/// @return NULL if `cmd` is NULL. Otherwise, a newly allocated command string.
+///         It must be freed with `xfree` when no longer needed.
+static char *shell_escape(const char *cmd)
+{
+  char *ncmd;
+
+  if (cmd == NULL) {
+    ncmd = NULL;
+  } else if (*p_sxq == NUL) {
+    ncmd = xstrdup(cmd);
+  } else {
+    const char *ecmd;
+    size_t ncmd_size;
+
+    if (*p_sxe != NUL && STRCMP(p_sxq, "(") == 0) {
+      ecmd = (char *)vim_strsave_escaped_ext((char_u *)cmd, p_sxe, '^', false);
+    } else {
+      ecmd = cmd;
+    }
+    ncmd_size = strlen(ecmd) + STRLEN(p_sxq) * 2 + 1;
+    ncmd = xmalloc(ncmd_size);
+
+    // When 'shellxquote' is '(', append ')'.
+    // When 'shellxquote' is '"(', append ')"'.
+    if (STRCMP(p_sxq, "(") == 0) {
+      snprintf(ncmd, ncmd_size, "(%s)", ecmd);
+    } else if (STRCMP(p_sxq, "\"(") == 0) {
+      snprintf(ncmd, ncmd_size, "\"(%s)\"", ecmd);
+    } else {
+      snprintf(ncmd, ncmd_size, "%s%s%s", p_sxq, ecmd, p_sxq);
+    }
```

Comment:
Since this PR is pretty small, I'm guessing fish support could be added without bloating it too much, but IMO: if you don't use fish or don't feel comfortable, placing a TODO where someone can find it later can suffice.